src/solver/simulation/timeseries-numbers.cpp

```diff
@@ -682,6 +682,39 @@ bool TimeSeriesNumbers::checkAllElementsIdenticalOrOne(std::vector<uint> w)
     return std::adjacent_find(w.begin(), first_one, std::not_equal_to<uint>()) == first_one;
 }
 
+bool TimeSeriesNumbers::CheckNumberOfColumns(const Study& study)
+{
+    std::map<const Antares::Data::TimeSeries::Numbers*, std::string> toCheck;
```

Comment:
Indexing on a pointer should be avoided: the key order will vary from one execution to another, leading to different orders in execution and messages.
It's better to have a vector of pairs instead (or small structs for naming stuff), because we don't care about finding based on the pointer.

Reply:
DONE